web/components/contract/quick-bet.tsx

```diff
@@ -0,0 +1,242 @@
+import { placeBet } from 'web/lib/firebase/api-call'
+import { getBinaryProb } from 'web/lib/firebase/contracts'
+import TriangleDownFillIcon from 'web/lib/icons/triangle-down-fill-icon'
+import TriangleFillIcon from 'web/lib/icons/triangle-fill-icon'
+import { Col } from '../layout/col'
+import { OUTCOME_TO_COLOR } from '../outcome-label'
+import { useSaveShares } from '../use-save-shares'
+import {
+  BinaryResolutionOrChance,
+  NumericResolutionOrExpectation,
+  FreeResponseResolutionOrChance,
+} from './contract-card'
+
+export function QuickBet(props: {
+  contract: Contract
+  setLiveUpdate: (liveUpdate: boolean) => void
+}) {
+  const { contract, setLiveUpdate } = props
+
+  const user = useUser()
+  const userBets = useUserContractBets(user?.id, contract.id)
+  const { yesFloorShares, noFloorShares, yesShares, noShares } = useSaveShares(
+    contract as FullContract<CPMM | DPM, Binary>,
+    userBets
+  )
+  // For some reason, Floor Shares are inverted for non-BINARY markets
+  const hasUpShares =
+    contract.outcomeType === 'BINARY' ? yesFloorShares : noFloorShares
+  const hasDownShares =
+    contract.outcomeType === 'BINARY' ? noFloorShares : yesFloorShares
+
+  const color = getColor(contract)
+
+  async function placeQuickBet(direction: 'UP' | 'DOWN') {
+    setLiveUpdate(true)
+    const betPromise = async () => {
+      const outcome = quickOutcome(contract, direction)
+      return await placeBet({
+        amount: 10,
+        outcome,
+        contractId: contract.id,
+      })
+    }
+    const shortQ = contract.question.slice(0, 20)
+    toast.promise(betPromise(), {
+      loading: `${formatMoney(10)} on "${shortQ}"...`,
+      success: `${formatMoney(10)} on "${shortQ}"...`,
+      error: (err) => `${err.message}`,
+    })
+  }
+
+  function quickOutcome(contract: Contract, direction: 'UP' | 'DOWN') {
+    if (contract.outcomeType === 'BINARY') {
+      return direction === 'UP' ? 'YES' : 'NO'
+    }
+    if (contract.outcomeType === 'FREE_RESPONSE') {
+      // TODO: Implement shorting of free response answers
+      if (direction === 'DOWN') {
+        throw new Error("Can't short free response answers")
+      }
+      return getTopAnswer(contract)?.id
+    }
+    if (contract.outcomeType === 'NUMERIC') {
+      // TODO: Ideally an 'UP' bet would be a uniform bet between [current, max]
+      throw new Error("Can't quick bet on numeric markets")
+    }
+  }
+
+  return (
+    <Col
+      className={clsx(
+        'relative -my-4 -mr-5 min-w-[6rem] justify-center gap-2 pr-5 pl-3 align-middle',
+        // Use this for colored QuickBet panes
+        // `bg-opacity-10 bg-${color}`
+        'bg-gray-50'
+      )}
+    >
+      {/* Up bet triangle */}
+      <div>
+        <div
+          className="peer absolute top-0 left-0 right-0 h-[50%]"
+          onClick={() => placeQuickBet('UP')}
+        ></div>
+        <div className="mt-2 text-center text-xs text-transparent peer-hover:text-gray-400">
```

Comment:
I think this should animate opacity on hover.

`transition-opacity ease-in duration-700 opacity-100 peer-hover:opacity-0`

Ditto for down arrow

Reply:
Appreciate the tip! I'd love to get a nice animation in there. When I tried this code, though, it seemed to run into weird corner cases where it'd flicker a lot (perhaps due to the way peer opacity works??)

I'll submit for now without the animation, but please please add it in afterwards!